lib/phoenix/transports/websocket.ex

```diff
@@ -13,36 +17,52 @@ defmodule Phoenix.Transports.WebSocket do
     ]
   end
 
-  def connect(%{method: "GET"} = conn, endpoint, handler, opts) do
+  def init(opts), do: opts
+
+  def call(%{method: "GET"} = conn, {endpoint, handler, opts}) do
     conn
-    |> Plug.Conn.fetch_query_params()
+    |> fetch_query_params()
     |> Transport.code_reload(endpoint, opts)
     |> Transport.transport_log(opts[:transport_log])
     |> Transport.force_ssl(handler, endpoint, opts)
```

Comment:
I think we no longer need this because the endpoint will always call `force_ssl` before we go down this path?

Reply:
It depends on where the `socket` clause ends up within the Endpoint, but yes you're correct.

Reply:
In fact, I realize now we can mostly get rid of this module if we move the socket upgrades to happen much later in the Plug stack, which is now possible. But that's a separate conversation...

For now, let's focus on getting `upgrade_adapter` without using Sock, so we can ship this in v1.7 and unblock a good part of Bandit integration, and then we can explore how to evolve all of this later!

Reply:
Sounds good! There are a couple of things above that I'm not sure I agree with just yet, but I think they'll be clearer if we can get all the other PRs in this stack done and dusted so we can focus on just this one more narrowly!